turbopack/crates/turbo-tasks-backend/src/utils/chunked_vec.rs

```diff
@@ -0,0 +1,76 @@
+pub struct ChunkedVec<T> {
+    chunks: Vec<Vec<T>>,
+}
```

Comment:
> worse average/amortized performance

Yes, but it should have a better worse case performance.

This is used for the transaction log and will be filled many entries. It's also accessed by every operation that modifies something in the system and need to be under a lock. So I thought it a large memmove could maybe stop the whole system for a while.

But later I discovered that the single mutex was a big performance problem anyway and added "sharding" on top of that. So that should reduce the problem a bit by making the vecs smaller and reduce the contention on that. Maybe we don't need it anymore...